azurerm/internal/services/network/network_interface_backend_address_pool_association_resource_test.go

```diff
@@ -136,59 +136,48 @@ func (t NetworkInterfaceBackendAddressPoolResource) Exists(ctx context.Context,
 	return utils.Bool(found), nil
 }
 
-func testCheckNetworkInterfaceBackendAddressPoolAssociationDisappears(resourceName string) resource.TestCheckFunc {
-	return func(s *terraform.State) error {
-		client := acceptance.AzureProvider.Meta().(*clients.Client).Network.InterfacesClient
-		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
-
-		// Ensure we have enough information in state to look up in API
-		rs, ok := s.RootModule().Resources[resourceName]
-		if !ok {
-			return fmt.Errorf("Not found: %s", resourceName)
-		}
+func (NetworkInterfaceBackendAddressPoolResource) destroy(ctx context.Context, client *clients.Client, state *terraform.InstanceState) error {
+	nicID, err := azure.ParseAzureResourceID(state.Attributes["network_interface_id"])
```

Comment:
Is there a reason we don't use the generated `NetworkInterfaceID` func in the service parse package here?

Reply:
(as below) we can for Destroy, can't for Exists